numba_dppy/target.py

```diff
@@ -73,7 +75,18 @@ def resolve_argument_type(self, val):
             ValueError: If the type of the Python value is not supported.
 
         """
-        if type(typeof(val)) is types.npytypes.Array:
+        try:
+            _type = type(typeof(val))
+        except ValueError:
+            # For arbitrary array that is not recognized by Numba,
+            # we will end up in this path. We check if the array
+            # has __sycl_usm_array_interface__ attribute. If yes,
+            # we create the necessary Numba type to represent it
+            # and send it back.
+            if has_usm_memory(val) is not None:
+                return suai_to_dppy_array_type(val)
+
+        if _type is types.npytypes.Array:
```

Comment:
As I understand this is the main part of the solution. Goal is to make DPPYTypingContext to resolve argument type to correct type for representing array with `__sycl_usm_array_interface__`.